tests/integration/mirror_push_test.go

```diff
@@ -119,3 +121,88 @@ func doRemovePushMirror(ctx APITestContext, address, username, password string,
 		assert.Contains(t, flashCookie.Value, "success")
 	}
 }
+
+func TestRepoSettingPushMirror(t *testing.T) {
+	defer tests.PrepareTestEnv(t)()
+
+	session := loginUser(t, "user2")
+
+	repoPrefix := "/user2/repo2"
+	repo2 := unittest.AssertExistsAndLoadBean(t, &repo_model.Repository{ID: 2})
+
+	defer test.MockVariableValue(&setting.Migrations.AllowedDomains, "127.0.0.1")()
+	assert.NoError(t, migrations.Init())
+	defer func() {
+		migrations.Init()
+	}()
+
+	// visit repository setting page
+	req := NewRequest(t, "GET", repoPrefix+"/settings")
+	resp := session.MakeRequest(t, req, http.StatusOK)
+	htmlDoc := NewHTMLParser(t, resp.Body)
+
+	defer func() {
+		// avoid dirty mirror data once test failure
+		repo_model.DeletePushMirrors(db.DefaultContext, repo_model.PushMirrorOptions{
+			RepoID: repo2.ID,
+		})
+	}()
+
+	onGiteaRun(t, func(t *testing.T, u *url.URL) {
```

Comment:
Why it needs onGiteaRun?

Reply:
Because `push_mirror_address` needs a URL.

Reply:
I do not see why it needs, and why the code was that complex.

8c27049649472d05e10f5f0c5b4fbc43e44bb485